### PR TITLE
LPD-25352

### PR DESCRIPTION
**Pull request by rijema**
Hi team, as the upgrade process is now considering the ctCollectionId and extending a base class created to the new tests with ctCollection, I would like to ask if maintaining testUpgradeAllDayCalendarBookingStartAndEndTime is necessary.

### CODE REVIEW

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (linolaoj) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:136]: Hi @rijema, should we use the expected prefix here?

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (linolaoj) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:50]: @rijema , we are standardizing the name of the upgrade now that we need to extend from `BaseCTUpgradeProcessTestCase`, please talk with @carolmariaabb to know about the standard.

**Review by linolaoj — CHANGES_REQUESTED** [review]

**Comment by marcelabc** [comment]
Hey @carolmariaabb 
Can you help double-check this PR? 
Since you did some of the tasks and reviewed others :smile: 
Thank you!

**Comment by MarinhoFeliphe** [comment]
Hey @rijema could you please push again? just to keep only your commits on this pull request

Thanks!

File: modules/apps/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/upgrade/v4_2_1/CalendarBookingUpgradeProcess.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/upgrade/v4_2_1/CalendarBookingUpgradeProcess.java:99]: @rijema we can remove the empty line here

File: modules/apps/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/upgrade/v4_2_1/CalendarBookingUpgradeProcess.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/upgrade/v4_2_1/CalendarBookingUpgradeProcess.java:39]: @rijema it should be `select ctCollectionId, calendarBookingId, companyId, userId, startTime, endTime` to follow the same order from [tables.sql](https://github.com/liferay/liferay-portal/blob/76041b96ce6b6cd7caf741a6da571a348a26801a/modules/apps/calendar/calendar-service/src/main/resources/META-INF/sql/tables.sql#L26-L44)

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [reply modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:50]: @rijema I will send a PR to Brian where I changed the pattern to:

`CalendarBookingCTUpgradeProcessTest.java`

See https://github.com/liferay-forms/liferay-portal/pull/2583.

Let's wait for his approval.

Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:50]: @rijema since we are going to rename it to something like `CalendarBookingCTUpgradeProcessTest`, can you create a different class instead of reusing this existing one? 

You can move the common code to a Util class `CalendarBookingUpgradeProcessTestUtil`

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:76]: @rijema sort alphabetically

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:133]: @rijema make it private

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [reply modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:136]: @rijema I don't think we should use the expected prefix

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:172]: @rijema this method is not necessary as it is only used once..

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:76]: @rijema make `createServiceContext` private

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [reply modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:76]: _serviceContext = ...
_upgradeDatabaseTestHelper = ..
_upgradeProcess = ..
_user = ...

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/CalendarBookingCTUpgradeProcessTest.java
Comment (rijema) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/CalendarBookingCTUpgradeProcessTest.java:83]: I would like to know if it's necessary to sort the Start and End here please

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/CalendarBookingCTUpgradeProcessTest.java
Comment (rijema) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/CalendarBookingCTUpgradeProcessTest.java:100]: Would be necessary to put this UpgradeStepRegistrator filter on Util module?

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (rijema) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:122]: I removed this bc I found a TestUtil class that would do the same.

File: modules/apps/calendar/calendar-test-util/src/main/java/com/liferay/calendar/test/util/CalendarBookingUpgradeProcessTestUtil.java
Comment (rijema) [inline modules/apps/calendar/calendar-test-util/src/main/java/com/liferay/calendar/test/util/CalendarBookingUpgradeProcessTestUtil.java:25]: I'm not sure if it's a ok practice to send a localService object as a param. It was necessary because I intend to update the current user.

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (rijema) [reply modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java]: As I sent to an util class, I had to leave it public. Is there any problem?

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (rijema) [reply modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java]: As I sent to an util class, I had to leave it public. Is there any problem?

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (rijema) [reply modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java]: I removed it . Not sure if you'll like, but I moved these instances to the createCalendar method.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by rijema** [comment]
ci:test:sf

File: modules/apps/calendar/calendar-test-util/src/main/java/com/liferay/calendar/test/util/CalendarBookingUpgradeProcessTestUtil.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test-util/src/main/java/com/liferay/calendar/test/util/CalendarBookingUpgradeProcessTestUtil.java:34]: @rijema do we still need this after doing `UserTestUtil.addUser(group.getGroupId());`?

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:52]: @rijema please remove the unused variable `_user`

File: modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java
Comment (carolmariaabb) [inline modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v4_2_1/test/UpgradeCalendarBookingTest.java:66]: @rijema `createCalendarBooking` does not have `user` as a parameter anymore

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by rijema** [comment]
I just compiled the test and everything went good, hopefully!

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by rijema** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:test:stable

**Comment by carolmariaabb** [comment]
@rijema 

```
junit.framework.AssertionFailedError: Please delete the unused testIntegrationImplementation dependencies in modules/apps/calendar/calendar-service/build.gradle
	at com.liferay.portal.modules.ModulesStructureTest._testGradleFile(ModulesStructureTest.java:1507)
```

**Comment by rijema** [comment]
Hi @carolmariaabb , updated. Thank you for letting me know. So, Ant format current branch or Ant all  scripts aren't supposed to remove this content as well, right?

**Comment by rijema** [comment]
ci:test:stable

**Comment by rijema** [comment]
I compiled the portal without the changes on build.gradle, the build finished and I was allowed to execute the test. Not sure why the IDE suggested to me at first place.

**Comment by carolmariaabb** [comment]
ci:forward:force

**Comment by carolmariaabb** [comment]
ci:test:sf

**Pull request closed without merging**
